cli/tests/pcluster/validators/test_secret_validators.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from pcluster.aws.common import AWSClientError
+from pcluster.validators.common import FailureLevel
+from pcluster.validators.secret_validators import ArnServiceAndResourceValidator, MungeKeySecretSizeAndBase64Validator
+from tests.pcluster.validators.utils import assert_failure_level, assert_failure_messages
+
+
+@pytest.mark.parametrize(
+    "arn, region, expected_service, expected_resource, expected_response",
+    [
+        (
+            "arn:aws:validate_service:us-west-2:123456789012:validate_resource",
+            "us-west-2",
+            "validate_service",
+            "validate_resource",
+            None,
+        ),
+        (
+            "arn:aws:s3:us-west-2:123456789012:bucket:my-bucket",
```

Comment:
if you are using _abstract_ services and resources as input (i.e. `validate_service` and `validate_resource`), please use abstract services also in your ARNs when you are checking for failures.

Reply:
Done!